### PR TITLE
feat: publish item to delphi API (issue-2)

### PR DESCRIPTION
**Pull request by brennan3**

### CODE REVIEW

File: .env.template [hunk 1]
```diff
@@ -1,3 +1,5 @@
 DELPHI_API_BASE_URL=
 DELPHI_READS_WEBHOOK_URL=
 DELPHI_READS_BOT_TOKEN=
+DELPHI_READS_API_KEY=
+DELPHI_POST_READS_ENDPOINT=
```
Comment (stocks29) [inline .env.template]: you think we need this? can we just use `DELPHI_API_BASE_URL`?

File: bots/reads.ts [hunk 2]
```diff
@@ -214,6 +216,30 @@ const displayMenu = async (ctx: ReadsContext) => {
...
+  const item = {
+    ...ctx.session.item,
+    tg: ctx.callbackQuery.from.username
+  };
+
+  const response = await fetch(postReadsUrl, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      'x-reads-bot-api-key': apiKey
```
Comment (stocks29) [inline bots/reads.ts]: should this just be `x-api-key` in case we want to use it for other bits in the future?

File: bots/reads.ts [hunk 3]
```diff
@@ -214,6 +216,30 @@ const displayMenu = async (ctx: ReadsContext) => {
   await ctx.reply('What would you like to do?', buttons);
 };
 
+const publish = async (ctx: ReadsContext, config: ReadsConfig) => {
+  const { delphiApi: { apiKey, postReadsEndpoint } } = config;
+  const postReadsUrl = delphiApiUrl(postReadsEndpoint, config);
+  const item = {
+    ...ctx.session.item,
```
Comment (stocks29) [inline bots/reads.ts]: maybe keep this nested under `item` key so we don't need to parse things apart if we add more fields?

File: bots/reads.ts [hunk 4]
```diff
@@ -214,6 +216,30 @@ const displayMenu = async (ctx: ReadsContext) => {
...
+      'Content-Type': 'application/json',
+      'x-reads-bot-api-key': apiKey
+    },
+    body: JSON.stringify(item)
+  });
+
+  const { ok } = await response.json();
+  
+  return ok
+    ? 'Item has been published, /new to publish another'
```
Comment (stocks29) [inline bots/reads.ts]: they can just paste a url directly to start the process again. prob can ditch the `/new` bit here.

File: bots/reads.ts [hunk 5]
```diff
@@ -214,6 +216,30 @@ const displayMenu = async (ctx: ReadsContext) => {
...
+    : 'Failed to publish item';
```
Comment (stocks29) [inline bots/reads.ts]: maybe `console.error` on non-200 response?

File: bots/reads.ts [hunk 6]
```diff
@@ -265,9 +291,9 @@ const handleNew = async (ctx: ReadsContext) => {
   await ctx.reply('what url do you want post?');
 };
 
-const handlePost = async (ctx: ReadsContext) => {
+const handlePost = async (ctx: ReadsContext, config: ReadsConfig) => {
```
Comment (stocks29) [inline bots/reads.ts:269]: I'd suggest having the `publish` function throw on error and generate the reply message here instead. this is the "controller" function whereas the `publish` function is really just a client function for sending the content to the endpoint.

File: server.ts [hunk 7]
```diff
@@ -32,7 +38,9 @@ const bot = readsBot(readsBotConfiguration);
 console.log('Configuration:', {
   DELPHI_API_BASE_URL,
   DELPHI_READS_WEBHOOK_URL,
+  DELPHI_POST_READS_ENDPOINT,
   DELPHI_READS_BOT_TOKEN: DELPHI_READS_BOT_TOKEN.replaceAll(/./g, '*'),
+  DELPHI_READS_API_KEY: DELPHI_READS_API_KEY.replaceAll(/./g, '*'),
```
Comment (stocks29) [inline server.ts]: maybe we have a `maskValue` helper since we now have two of these?

File: bots/reads.ts [hunk 8]
```diff
@@ -265,9 +291,9 @@ const handleNew = async (ctx: ReadsContext) => {
...
   ensureLinkSet(ctx, async () => {
-    await ctx.reply('TODO: implement me');
+    await ctx.reply(await publish(ctx, config));
```
Comment (stocks29) [inline bots/reads.ts]: after posting successfully we should reset the state here:

```ts
// after publish
resetState(ctx);
```

**Review by stocks29 — APPROVED** [review]

**Comment by DelphiGitHubMaintainer** [comment]
fixes https://github.com/delphidigital/delphi-tg-bots/issues/10

**Pull request merged**